src/routes/verify.ts

```diff
@@ -6,41 +6,40 @@ import { generateReceiptSecret } from '../util/crypto'
 
 export const RECEIPT_LENGTH_BASE64 = 80
 
-const verifyReceipt = () => async (ctx: Koa.Context, next: Koa.Next) => {
+export const router = new Router()
+
+router.post('/verify', async (ctx: Koa.Context) => {
   const body = await raw(ctx.req, {
     limit: RECEIPT_LENGTH_BASE64
   })
 
   let receipt: Receipt
   try {
     const receiptBytes = Buffer.from(body.toString(), 'base64')
-    ctx.state.receipt = verifyReceiptBytes(receiptBytes, (decoded: ReceiptWithHMAC) => {
+    receipt = verifyReceiptBytes(receiptBytes, (decoded: ReceiptWithHMAC) => {
       return generateReceiptSecret(ctx.config.receiptSeed, decoded.nonce)
     })
   } catch (error) {
     ctx.throw(400, error.message)
   }
 
   try {
-    ctx.state.receiptValue = await ctx.redis.getReceiptValue(ctx.state.receipt)
+    const { value, spspEndpoint, spspId } = await ctx.redis.getReceiptValue(receipt)
+    if (value!.isZero()) {
```

Comment:
maybe `if (!value or value.isZero())`? I'm thinking that if value is undefined we might want to still throw an error?

Reply:
https://github.com/coilhq/receipt-verifier/pull/25/commits/799f67b2603b5d5e9b695950bef6d5d42e474215